Utilities/bootstrap

```diff
@@ -481,7 +481,7 @@ def install_binary(args, binary, destination, destination_is_directory=True, ign
 # Build functions
 # -----------------------------------------------------------
 
-def build_with_cmake(args, cmake_args, ninja_args, source_path, build_dir):
+def build_with_cmake(args, cmake_args, ninja_args, source_path, build_dir, cmake_env = []):
```

Comment:
nit: `args, cmake_args,  cmake_env = [], ninja_args, source_path, build_dir`

Reply:
not possible afaik, python doesn't allow non-default args after args with default values